docs/rfcs/0561-list-metadata-reuse.md

```diff
@@ -0,0 +1,131 @@
+- Proposal Name: `list_metadata_reuse`
+- Start Date: 2022-08-23
+- RFC PR: [datafuselabs/opendal#561](https://github.com/datafuselabs/opendal/pull/561)
+- Tracking Issue: None
+
+# Summary
+
+Extend `DirEntry` with more `metadata` fields.
+
+# Motivation
+
+Users may expect to browse metadata of some directories' child files and directories. `list()` seems to be an ideal way to complete this job. 
+
+Thus, they start iterating on it, but soon they realized the `DirEntry`, which could only offer the name (or path, more precisely) and access mode of the object.
+
+So they have to call `metadata()` for each name they extracted from the iterator.
+
+The final example looks like:
+
+```rust
+use anyhow::Result;
+use futures_lite::StreamExt;
+use opendal::ops::OpList;
+use opendal::ops::OpStat;
+use opendal::services::gcs::Builder;
+use opendal::Accessor;
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    let list_op = OpList::new("/path/to/dir")?;
+    let accessor = Builder::default()
+        .bucket("example")
+        .root("/example/root/")
+        .credential("example-credential")
+        .build()?;
+
+    // here is a network request
+    let mut dir_stream = accessor.list(&list_op).await?;
```

Comment:
All our public APIs starts from `Operator`. Thus, our examples should also use `op` instead. For example: https://opendal.databend.rs/rfcs/0413-presign.html

We can digging into Accessor's API inside `Reference-level explanation`